### PR TITLE
Fix chained ObserverHandler typing

### PR DESCRIPTION
**Pull request by frmdstryr**
mypy doesn't like chained observers like enaml uses.

### CODE REVIEW

File: examples/api/observe.py [hunk 1]
```diff
@@ -37,6 +37,14 @@ def another_random_name(self, change: ChangeDict) -> None:
...
 
 
+class Employee(Person):
+    title = Str()
+
+    @observe("title")
+    def on_title_change(self, change: ChangeDict) -> None:
+        super().any_name_i_want(change)
+
+
```
Comment (MatthieuDartiailh) [inline examples/api/observe.py:47]: What is this supposed to demonstrate ?

File: atom/atom.py [hunk 2]
```diff
@@ -117,7 +117,11 @@ def __init__(
 
     def __call__(
         self,
-        func: Union[Callable[[ChangeDict], None], Callable[[T, ChangeDict], None]],
+        func: Union[
```
Comment (MatthieuDartiailh) [inline atom/atom.py:120]: Looking at the implementation I don't see how it works. Could you point me to an example in the enaml codebase ?

Comment (frmdstryr) [reply atom/atom.py:120]: If you uncomment the "ObserverHandler" then run `mypy examples` it fails when calling the super method.

**Comment by frmdstryr** [comment]
Well this seems like a bug in mypy. It doesn't know how to unroll the observe decorators.

**Comment by frmdstryr** [comment]
https://www.python.org/dev/peps/pep-0612/

File: atom/atom.py [hunk 3]
```diff
@@ -117,7 +117,11 @@ def __init__(
...
+            Callable[[ChangeDict], None],
+            Callable[[T, ChangeDict], None],
+            ChangeDict,  # TODO: Workaround for mypy
```
Comment (MatthieuDartiailh) [inline atom/atom.py:123]:
```suggestion
            # AtomMeta will replace ObserveHandler in the body of an atom 
            # class allowing to access it for example in a subclass. We lie here by
            # giving ObserverHandler.__call__ a signature compatible with an 
            # observer to mimic this behavior. 
            ChangeDict,
```

**Comment by MatthieuDartiailh** [comment]
I am not sure this really helps us here since the point is that the returned value won't be what is bound to the name later on. 

It iseasier to simply lie here by either saying that the we return the input callable using a TypeVar and use a type: ignore in the return of either observe or observe handler, or lie about the signature of `ObserveHandler.__call__` like you did. The nice part about your lie is that it does not requires a type: ignore. See my suggestion and I think we are good to go.

**Pull request merged**